src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,39 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            return new BulldozerProducer().get();
+        } else if (type == Truck.class) {
+            return new TruckProducer().get();
+        } else if (type == Excavator.class) {
+            return new ExcavatorProducer().get();
+        }
+        return new ArrayList<>();
```

Comment:
Consider creating a local variable for the `machines` list in the `getAll` method to avoid duplication and direct return of the `get` method call. This will make it easier to manage the list before returning, if needed in the future.